submission/submission-core/src/main/kotlin/ac/uk/ebi/biostd/submission/domain/request/SubmissionRequestReleaser.kt

```diff
@@ -46,10 +48,16 @@ class SubmissionRequestReleaser(
         version: Int,
         processId: String,
     ) {
-        rqtService.onRequest(accNo, version, FILES_COPIED, processId, {
-            if (it.submission.released) releaseRequest(accNo, it)
+        rqtService.onRequest(accNo, version, FILES_COPIED, processId) {
+            if (it.submission.released) {
+                releaseRequest(accNo, it)
+            } else {
+                val current = queryService.findExtByAccNo(accNo, includeFileListFiles = false)
```

Comment:
it looks too much to query all the submission info (except files) for only determinaning if submission has a previous version. We can query only that or use CoreInfo queery operation.